lms/djangoapps/instructor/views/api.py

```diff
@@ -1051,7 +1060,7 @@ def get_coupon_codes(request, course_id):  # pylint: disable=unused-argument
     return instructor_analytics.csvs.create_csv_response('Coupons.csv', header, data_rows)
 
 
-def save_registration_code(user, course_id, mode_slug, invoice=None, order=None):
+def save_registration_code(user, course_id, mode_slug, invoice=None, order=None, invoice_item=None):
```

Comment:
I second this comment.  Maybe change the method signature to accept `invoice_item`, then get the invoice using `invoice_item.invoice`?